File: 2825/Solution.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        for(int i = 0, j = 0; i < str1.length(); i++){
            char c = str1.charAt(i);
            char x = (char) (c + 1);
            if(x == 123) x = 'a';
            if(str2.charAt(j) == c || str2.charAt(j) == x) {
                j++;
                if(j == str2.length()) return true;
            } 
        }
        return false;
    }
}

File: 2825/Solution.go
func canMakeSubsequence(str1 string, str2 string) bool {
    i, j := 0, 0
    for i < len(str1) && j < len(str2) {
        x := str1[i] + 1
        if str1[i] == 'z' {
            x = 'a'
        }

        if str1[i] == str2[j] || str2[j] == x {
            i++
            j++
        }else {
            i++
        }
    }
    return j == len(str2)

}
